core/src/main/java/org/infinispan/stream/impl/DistributedCacheStream.java

```diff
@@ -106,35 +106,11 @@
     * @param distributedBatchSize default size of distributed batches
     * @param executor executor to be used for certain operations that require async processing (ie. iterator)
     */
```

Comment:
We should probably update the javadoc to explain the purpose of `toKeyFunction` and maybe explain how it's required for tracking duplicates etc.